packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -709,8 +713,12 @@ async function createComponentTreeInternal({
         {layerAssets}
         <OutletBoundary>
           <MetadataOutlet ready={getViewportReady} />
+          {/* Blocking metadata outlet */}
           <MetadataOutlet ready={getMetadataReady} />
-          <StreamingMetadataOutlet />
+          {/* Streaming metadata outlet */}
+          {actualSegment !== DEFAULT_SEGMENT_KEY ? (
```

Comment:
could you clarify why we only need this special case for the streaming metadata?

Reply:
same as what we've done for metadat rendering here, I'll add some comments 
https://github.com/vercel/next.js/blob/0dc2c011cc7892d95749af83cbd5bcb0c6a3f475/packages/next/src/server/app-render/create-component-tree.tsx#L398-L400